### PR TITLE
UBERF-5564: rework groupping and support PersonAccount 

### PR DESCRIPTION
**Pull request by ThetaDR**
Move aggregation. Make it customizable for classes

### CODE REVIEW

File: models/core/src/security.ts [hunk 1]
```diff
@@ -154,7 +154,7 @@ export class TSpacesTypeData extends TSpace implements RolesAssignment {
 }
 
 @Model(core.class.Account, core.class.Doc, DOMAIN_MODEL)
-@UX(core.string.Account)
+@UX(core.string.Account, undefined, undefined, 'name')
```
Comment (BykhovDenis) [inline models/core/src/security.ts:169]: We don't have name in account, why we need it here?

File: plugins/view-resources/src/components/list/ListCategories.svelte [hunk 2]
```diff
@@ -353,126 +353,138 @@
...
+    ...docKeys,
+    [groupByKey]:
+      typeof category === 'object'
+        ? category.name !== undefined
+          ? { $in: category.values.flatMap((x) => x._id) }
+          : resultQuery[groupByKey]?.$in?.length !== 0
+            ? undefined
+            : []
+        : category
+  }}
```
Comment (BykhovDenis) [inline plugins/view-resources/src/components/list/ListCategories.svelte]: Better extract

File: plugins/view-resources/src/components/list/ListCategory.svelte [hunk 3]
```diff
@@ -109,12 +109,14 @@
...
   $: if (lastLevel) {
     void limiter.add(async () => {
+      // console.log('category Query', _class, { ...resultQuery, ...docKeys })
       loading = docsQuery.query(
         _class,
         { ...resultQuery, ...docKeys },
         (res) => {
           items = res
           loading = false
+          console.log({ docKeys }, res)
```
Comment (BykhovDenis) [inline plugins/view-resources/src/components/list/ListCategory.svelte]: extra log

File: plugins/view-resources/src/components/list/ListCategory.svelte [hunk 4]
```diff
@@ -109,12 +109,14 @@
 
   $: if (lastLevel) {
     void limiter.add(async () => {
+      // console.log('category Query', _class, { ...resultQuery, ...docKeys })
```
Comment (BykhovDenis) [inline plugins/view-resources/src/components/list/ListCategory.svelte]: extra log

**Review by BykhovDenis — DISMISSED** [review]

File: plugins/contact-resources/src/utils.ts [hunk 5]
```diff
@@ -415,3 +423,109 @@ export async function channelTitleProvider (client: Client, ref: Ref<Channel>, d
...
+      personAccountList.push(personAccount)
+      usedSpaces.add(personAccount.space)
+    }
+  }
+
+  for (const personAccount of personAccountList) {
+    if (personAccount !== undefined) {
+      let fst = personMap.get(personAccount.person)
+      if (fst === undefined) {
+        const components = mgr
```
Comment (haiodo) [inline plugins/contact-resources/src/utils.ts]: why it named components?

Comment (ThetaDR) [reply plugins/contact-resources/src/utils.ts]: Typo, fixed

File: models/core/src/security.ts (hunk 1 above)
Comment (ThetaDR) [reply models/core/src/security.ts:169]: Required for enabling search

**Review by BykhovDenis — DISMISSED** [review]

**Review by haiodo — APPROVED** [review]

**Pull request merged**